cmd/krew/cmd/root.go

```diff
@@ -104,9 +107,31 @@ func preRun(cmd *cobra.Command, _ []string) error {
 			klog.Warningf("You may need to clean them up manually. Error: %v", err)
 		}
 	}
+
+	go func() {
+		if _, ok := os.LookupEnv("KREW_NO_UPGRADE_CHECK"); ok {
+			return
+		}
+		if msg := assertion.CheckVersion(paths.BasePath()); msg != "" {
+			notifications <- msg
```

Comment:
we expect 1 result.
so try avoid channels if you can. we can simply check if string is empty or not.

Reply:
I was planning ahead of collecting messages to be shown on command completion. But you're right that for now a simple global will do the job.
Although we should not run our tests with `-race` anymore ;)